src/commands/up.rs

```diff
@@ -66,10 +70,21 @@ pub struct UpCommand {
         name = BINDLE_PASSWORD,
         long = "bindle-password",
         env = BINDLE_PASSWORD,
-        requires = BINDLE_USERNAME
+        requires = BINDLE_USERNAME,
+        hide = true,
     )]
     pub bindle_password: Option<String>,
 
+    /// Reference to run the application from a container registry.
+    #[clap(
+        name = REFERENCE_OPT,
+        long = "reference",
```

Comment:
To me, this name is very vague now that it's decoupled from the `oci/registry` noun context.  I am not sure what is better: `registry-reference` is certainly too long.

Reply:
Possibly terrible suggestion: `--from-registry ghcr.io/username/app:v1`

Reply:
`--open-container-initiative-artifact-reference`

Reply:
Is there resistance to simply `--oci` ?

Reply:
Otherwise `--from-registry` is fine with me.

Reply:
We just renamed `spin oci` to `spin registry`. :shrug:

Reply:
Yeah `spin up --from-registry` reads fine to me.

Reply:
Meh, but I could imagine a future where we have a `--warg` option.

Alternatively we could make `spin up` accept a positional argument and infer the source from the scheme....

Reply:
As a developer tool I think it would be fine for `spin up` to detect the registry type. We can always add e.g. `--registry-type warg` (defaulting to `auto`) later too.

Reply:
Changed to `--from-registry`.